File: j-child/src/test/java/net/namingcrisis/sweetchildomine/jchild/util/AssertTest.java
package net.namingcrisis.sweetchildomine.jchild.util;

import org.junit.Test;

import static net.namingcrisis.sweetchildomine.jchild.util.TestUtil.mustFail;

public final class AssertTest {

  @Test
  public void assertTrue() {
    Assert.assertTrue(true, () -> {
      throw new AssertionError("Never expected to be called!");
    });

    mustFail(
      () -> Assert.assertTrue(false, () -> "Failed condition"),
      "Failed condition"
    );
  }

  @Test
  public void assertFalse() {
    Assert.assertFalse(false, () -> {
      throw new AssertionError("Never expected to be called!");
    });

    mustFail(
      () -> Assert.assertFalse(true, () -> "Failed condition"),
      "Failed condition"
    );
  }
}

File: ts-child/test/util/Assert.test.ts
import { Assert } from "@app/util/Assert";

describe("Assert", () => {
  test("assertTrue", () => {
    Assert.assertTrue(true, () => { throw new Error("Must never be called!"); });
    expect(
      () => Assert.assertTrue(false, () => "Failed condition")
    ).toThrow('Failed condition');
  });

  test("assertFalse", () => {
    Assert.assertFalse(false, () => { throw new Error("Must never be called!"); });
    expect(
      () => Assert.assertFalse(true, () => "Failed condition")
    ).toThrow('Failed condition');
  });
});
